### PR TITLE
Updating Roslyn references

### LINKED ISSUE #5343: Update Functions V2 to roslyn 3.4.0
Functions V2 is not running the latest version of Roslyn. We know there are still reliability issues with the currently referenced version (see https://github.com/Azure/azure-functions-host/issues/999). We should take the update and then continue to monitor.

### PR DESCRIPTION
**Pull request by fabiocav**
Resolves #5343

### CODE REVIEW

File: src/WebJobs.Script/Description/DotNet/Compilation/CSharp/Analyzers/AsyncVoidAnalyzer.cs [hunk 1]
```diff
@@ -27,6 +27,8 @@ public AsyncVoidAnalyzer()
 
         public override void Initialize(AnalysisContext context)
         {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
             context.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.Method);
```
Comment (brettsam) [inline src/WebJobs.Script/Description/DotNet/Compilation/CSharp/Analyzers/AsyncVoidAnalyzer.cs:32]: Any risk with these changes? Looks like a little more than just a nuget update...

Comment (fabiocav) [reply src/WebJobs.Script/Description/DotNet/Compilation/CSharp/Analyzers/AsyncVoidAnalyzer.cs:32]: These are the same changes I had to make in the 3.0 branch when taking in the new major versions.

Comment (brettsam) [reply src/WebJobs.Script/Description/DotNet/Compilation/CSharp/Analyzers/AsyncVoidAnalyzer.cs:32]: Oh was that already done and released? Didn't realize this was related. If so, looks good.

**Review by brettsam — DISMISSED** [review]

**Comment by fabiocav** [comment]
Review state was lost after rebase, merging, but let me know if you have any concerns.

**Pull request merged**